### PR TITLE
docs: fix some dead links

### PR DESCRIPTION
**Pull request by aanm**
Replaced some dead links with alive ones.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: Documentation/gettingstarted/requirements-rke.rst [hunk 1]
```diff
@@ -4,10 +4,10 @@ perform the following steps:
 .. note::
 
    If you are using RKE2, Cilium has been directly integrated. Please see
-   `Using Cilium <https://docs.rke2.io/install/network_options/#using-cilium-instead-of-canal>`_
+   `Using Cilium <https://docs.rke2.io/install/network_options/#using-cilium-or-calico-instead-of-canal>`_
    in the RKE2 documentation. You can use either method.
 
-   https://docs.rke2.io/install/network_options/#using-cilium-instead-of-canal
+   https://docs.rke2.io/install/network_options/#using-cilium-or-calico-instead-of-canal
```
Comment (qmonnet) [inline Documentation/gettingstarted/requirements-rke.rst]: Let's remove this one? It is identical to the link in the paragraph above.

**Review by qmonnet — CHANGES_REQUESTED** [review]
I haven't checked the old links, but the new ones are all valid, thanks!

One suggestion below.

How did you find the dead links, did you run Sphinx's check feature?

**Review by qmonnet — APPROVED** [review]
Thanks!

**Comment by aanm** [comment]
> I haven't checked the old links, but the new ones are all valid, thanks!
> 
> One suggestion below.
> 
> How did you find the dead links, did you run Sphinx's check feature?

ah I forgot to reply, yes. With the current commented that checks for dead links

**Pull request merged**